apps/omg_child_chain/test/omg_child_chain/event_fetcher_test.exs

```diff
@@ -0,0 +1,497 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+defmodule OMG.ChildChain.EventFetcherTest do
+  use ExUnit.Case, async: true
+
+  alias OMG.ChildChain.EventFetcher
+  alias OMG.Eth.RootChain.Abi
+
+  setup do
+    table = :ets.new(String.to_atom("test-#{:rand.uniform(1000)}"), [:bag, :public, :named_table])
+    event_fetcher_name = String.to_atom("test-#{:rand.uniform(1000)}")
+
+    start_supervised(
+      {EventFetcher,
+       name: event_fetcher_name,
+       contracts: %{},
+       ets_bucket: table,
+       events: [
+         [name: :deposit_created, enrich: false],
+         [name: :exit_started, enrich: true],
+         [name: :in_flight_exit_input_piggybacked, enrich: false],
+         [name: :in_flight_exit_output_piggybacked, enrich: false],
+         [name: :in_flight_exit_started, enrich: true]
+       ]}
+    )
+
+    {:ok, %{event_fetcher_name: event_fetcher_name, table: table}}
+  end
+
+  test "the performance of event retrieving", %{table: table, event_fetcher_name: event_fetcher_name, test: test_name} do
+    defmodule test_name do
+      alias OMG.ChildChain.EventFetcherTest
+
+      def get_ethereum_events(_from_block, to_block, _signatures, _contracts) do
+        deposits = for n <- 1..10_000, do: EventFetcherTest.deposit_created_log(n)
+        {:ok, [EventFetcherTest.in_flight_exit_input_piggybacked_log(to_block) | deposits]}
+      end
+
+      def get_call_data(_tx_hash) do
+        {:ok, EventFetcherTest.start_standard_exit_log()}
+      end
+    end
+
+    from_block = 1
+    to_block = 80_000
+    :sys.replace_state(event_fetcher_name, fn state -> Map.put(state, :rpc, test_name) end)
+    events = event_fetcher_name |> :sys.get_state() |> Map.get(:events)
+    {time, _value} = :timer.tc(EventFetcher, :deposit_created, [event_fetcher_name, from_block, to_block])
+    assert Enum.count(:ets.tab2list(table)) == Enum.count(events) * 80_000
+    duration = time / 1_000_000
+    assert duration < 5
+  end
+
+  test "that events and event signatures are correctly initialized ", %{event_fetcher_name: event_fetcher_name} do
+    assert event_fetcher_name |> :sys.get_state() |> Map.get(:events) == [
```

Comment:
I'd be good to test without going through the internals. Is the tested behavior here that none of those events is missed? or that `enrich: true` has an expected effect? or sth else?